C-Plus-Plus/dp/numberOf_subsets_with_given_diff.cpp

```diff
@@ -0,0 +1,61 @@
+// Program to find number of subsets with given difference between their sums
+// Uses dynamic programming to solve the problem. We'll use the existing problem that counts subsets with a given sum.
+#include <bits/stdc++.h>
+using namespace std;
+int dp[20001][20001];
+// counting number of subsets
+int countSubsets(int arr[], int n, int sum)
+{
+	if (dp[n][sum] != -1)
+		return dp[n][sum];
+	if (sum == 0)
+		return 1;
+	if (n == 0)
+		return 0;
+	if (arr[n - 1] <= sum)
+		return dp[n][sum] = countSubsets(arr, n - 1, sum - arr[n - 1]) + countSubsets(arr, n - 1, sum);
+	return dp[n][sum] = countSubsets(arr, n - 1, sum);
+}
+
+int subsetsDiff(int arr[], int n, int diff)
+{
+	int arr_sum = 0;
+	for (int i = 0; i < n; i++)
+		arr_sum += arr[i];
+	/*Let there be two subsets with sum s1 and s2 respectively
+	s1 + s2 = arr_sum
+	s1 - s2 = diff
+	so solving the above equations, we get:
+	s1 = (diff + arr_sum)/2
+	Therefore, we will find number of subsets with sum as s1 */
```

Comment:
Remove that redundant comments

Reply:
This comment explains the main logic of the code which is making it simpler. Should I remove this?

Reply:
Okay then no need to remove. Use multiline comments. Don't use single line comments.